File: SecondSem/CN/LAB/Java/ConnectionLessSimplex_2/UDPServer.java
import java.net.DatagramPacket;
import java.net.DatagramSocket;

public class UDPServer {
    public static void main(String[] args) {
        try {

            DatagramSocket serverSocket = new DatagramSocket(9876);

            byte[] receiveData = new byte[1024];

            System.out.println("UDP Server is running...");

            while (true) {
                // Create a DatagramPacket to receive data
                DatagramPacket receivePacket = new DatagramPacket(receiveData, receiveData.length);

                // Receive data from the client
                serverSocket.receive(receivePacket);

                // Extract the received message
                String message = new String(receivePacket.getData(), 0, receivePacket.getLength());
                System.out.println("Received from client: " + message);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: SecondSem/CN/LAB/Python/ConnectionLessSimplex_2/UDPserver.py
import socket


class UDPServer:
    def __init__(self, port=9876):
        self.server_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
        self.server_socket.bind(("", port))
        print("UDP Server is running...")

    def run(self):
        while True:
            # Receive data from the client
            receive_data, client_address = self.server_socket.recvfrom(1024)

            # Extract the received message
            message = receive_data.decode("utf-8")
            print(f"Received from client: {message}")


if __name__ == "__main__":
    server = UDPServer()
    server.run()
